dao/src/test/java/se/fortnox/reactivewizard/db/DbProxyTest.java

```diff
@@ -470,25 +529,29 @@ public void shouldFailIfMinimumAffectedRowsNotReachedForVoidReturnType() throws
     }
 
     @Test
-    public void shouldSupportBackpressure() throws SQLException, InterruptedException {
+    public void shouldSupportBackpressure() throws SQLException {
         mockDb.addRowColumn(1, 1, "name", String.class, "row1");
         mockDb.addRowColumn(2, 1, "name", String.class, "row2");
-        TestSubscriber<DbTestObj> testSubscriber = new TestSubscriber<>(0);
-        dbProxyTestDao.select("mykey").subscribe(testSubscriber);
-
-        testSubscriber.requestMore(1);
-        Thread.sleep(100);
-        testSubscriber.assertValueCount(1);
-        testSubscriber.assertNoTerminalEvent();
 
-        verify(mockDb.getConnection()).close();
-        verify(mockDb.getPreparedStatement()).close();
-        verify(mockDb.getResultSet()).close();
+        StepVerifier.create(dbProxyTestDao.selectFlux("mykey"))
+            .thenRequest(1)
+            .thenAwait(ofMillis(100))
+            .expectNextCount(1)
+            .then(this::assertMockDbClosed)
+            .thenRequest(2)
+            .thenAwait(ofMillis(100))
+            .expectNextCount(1)
+            .verifyComplete();
+    }
 
-        testSubscriber.requestMore(2);
-        Thread.sleep(100);
-        testSubscriber.assertValueCount(2);
-        testSubscriber.assertCompleted();
+    private void assertMockDbClosed() {
+        try {
+            verify(mockDb.getConnection()).close();
+            verify(mockDb.getPreparedStatement()).close();
+            verify(mockDb.getResultSet()).close();
+        } catch (SQLException e) {
```

Comment:
When does this happen?

Reply:
This happens in _shouldSupportBackpressure_ test. I just moved the code to another method. Other than this test, I check that the connection closes in some of the new tests added in this PR.